### PR TITLE
[UnitTest]Enable ipu utest

### PR DESCRIPTION
**Pull request by xuhongyao**
add unit test cases to IPU test

### CODE REVIEW

**Comment by weimingzha0** [comment]
needs rebased

**Comment by weimingzha0** [comment]
there are conflicts. could you rebase?

File: ODLA/platforms/odla_popart/odla_compute.cc [hunk 1]
```diff
@@ -1,7 +1,6 @@
-//===- odla_compute.cc ----------------------------------------------------===//
+//===- odla_popart.cc -----------------------------------------------------===//
```
[the remark this answers is not in the document]
Comment (xuhongyao) [reply ODLA/platforms/odla_popart/odla_compute.cc:1]: ok

File: ODLA/platforms/odla_popart/odla_compute.cc [hunk 2]
```diff
@@ -69,7 +68,9 @@ std::unique_ptr<popart::SessionOptions> SessionOptions() {
   auto opts =
       std::unique_ptr<popart::SessionOptions>(new popart::SessionOptions());
   opts->virtualGraphMode = popart::VirtualGraphMode::Auto;
+  // opts->enablePipelining = true;
```
[the remark this answers is not in the document]
Comment (xuhongyao) [reply ODLA/platforms/odla_popart/odla_compute.cc]: yes

File: ODLA/platforms/odla_popart/common.cc [hunk 3]
```diff
@@ -1,7 +1,6 @@
 //===- odla_common.cc -----------------------------------------------------===//
 //
 // Copyright (C) 2019-2020 Alibaba Group Holding Limited.
```
Comment (weimingzha0) [inline ODLA/platforms/odla_popart/common.cc:3]: keep the copyright

File: ODLA/platforms/odla_popart/odla_compute.cc [hunk 4]
```diff
@@ -1,7 +1,6 @@
-//===- odla_compute.cc ----------------------------------------------------===//
+//===- odla_popart.cc -----------------------------------------------------===//
 //
 // Copyright (C) 2019-2020 Alibaba Group Holding Limited.
-// Copyright (c) 2020 Graphcore Ltd. All rights reserved.
```
Comment (weimingzha0) [inline ODLA/platforms/odla_popart/odla_compute.cc:4]: keep?

File: ODLA/platforms/odla_popart/odla_compute.cc [hunk 5]
```diff
@@ -69,7 +68,9 @@ std::unique_ptr<popart::SessionOptions> SessionOptions() {
   auto opts =
       std::unique_ptr<popart::SessionOptions>(new popart::SessionOptions());
   opts->virtualGraphMode = popart::VirtualGraphMode::Auto;
+  // opts->enablePipelining = true;
   opts->enableStochasticRounding = true;
+  // TODO(unknown) add more opts
```
Comment (weimingzha0) [inline ODLA/platforms/odla_popart/odla_compute.cc]: remove?

File: ODLA/platforms/odla_popart/odla_ops.cc [hunk 6]
```diff
@@ -1,7 +1,6 @@
-//===- odla_ops.cc --------------------------------------------------------===//
+//===- odla_popart.cc -----------------------------------------------------===//
 //
 // Copyright (C) 2019-2020 Alibaba Group Holding Limited.
-// Copyright (c) 2020 Graphcore Ltd. All rights reserved.
```
Comment (weimingzha0) [inline ODLA/platforms/odla_popart/odla_ops.cc:4]: keep

File: ODLA/platforms/odla_popart/odla_popart.h [hunk 7]
```diff
@@ -1,7 +1,6 @@
-//===- odla_popart.h ------------------------------------------------------===//
+//===- odla_popart.cc -----------------------------------------------------===//
 //
 // Copyright (C) 2019-2020 Alibaba Group Holding Limited.
-// Copyright (c) 2020 Graphcore Ltd. All rights reserved.
```
Comment (weimingzha0) [inline ODLA/platforms/odla_popart/odla_popart.h:4]: keep

File: ODLA/platforms/odla_popart/odla_compute.cc [hunk 8]
```diff
@@ -1,4 +1,4 @@
-//===- odla_compute.cc ----------------------------------------------------===//
+//===- odla_compute.cc ------------------------------------------------===//
```
Comment (weimingzha0) [inline ODLA/platforms/odla_popart/odla_compute.cc:1]: check if 80 column

**Review by weimingzha0 — APPROVED** [review]

**Pull request merged**